### ISSUE TITLE
Add the benchmark for ResNet50

### ISSUE THREAD

**Issue by kiya00** [Lightning-AI/lightning-thunder#443]
## 🚀 Feature

Add the benchmark for ResNet50

cc @crcrpar @IvanYashchuk @jjsjann123

**Comment by kiya00** [comment]
Modifications of the original resnet50 in torchvision:
1. replace `out += identity` with `out = out + identity`
2. `ReLU(replace=False)`
3. set `num_batches_tracked = None` for BatchNorm (a workaround since `add_` is utilized in `nn.BatchNorm3d` when `num_batches_tracked` is not None).

**Comment by kiya00** [comment]
Modifications of the original resnet50 in torchvision:

replace out += identity with out = out + identity
ReLU(replace=False)
set num_batches_tracked = None for BatchNorm (a workaround since add_ is utilized in nn.BatchNorm3d when num_batches_tracked is not None).

needs: #365
